layers/+lang/sql/funcs.el

```diff
@@ -16,3 +16,45 @@
    spacemacs-sql-startable (remove-if-not
                             (lambda (product) (sql-get-product-feature (car product) :sqli-program))
                             sql-product-alist)))
+
+(defun spacemacs//sql-backend ()
+  "Returns selected backend."
+  (if sql-backend
+      sql-backend
+    (cond
+     ((configuration-layer/layer-used-p 'lsp) 'lsp)
+     (t 'company-sql))))
+
+(defun spacemacs//sql-setup-company ()
+  "Conditionally setup company based on backend."
+  (pcase (spacemacs//sql-backend)
+    ('company-sql (spacemacs|add-company-backends
+                    :backends company-capf
+                    :modes sql-mode))
+    ;; Activate lsp company explicitly to activate
+    ;; standard backends as well
+    (`lsp (spacemacs|add-company-backends
+            :backends company-capf
+            :modes sql-mode))))
+
+(defun spacemacs//sql-setup-backend ()
+  "Conditionally setup sql backend."
+  (pcase (spacemacs//sql-backend)
+    (`lsp (spacemacs//sql-setup-lsp))))
+
+;; lsp
+
+(defun spacemacs//sql-setup-lsp ()
```

Comment:
Do we need to require lsp-sqls directly? I would expect that this is loaded by lsp automatically.
Do we need to setup the sql variables for every buffer aka are they buffer local? If not I would prefer to move them to sql package init.